src/live_restore/live_restore_fs.c

```diff
@@ -1524,13 +1526,19 @@ __live_restore_fs_atomic_copy_file(WT_SESSION_IMPL *session, WTI_LIVE_RESTORE_FS
  */
 static int
 __live_restore_setup_lr_fh_file_data(WT_SESSION_IMPL *session, WTI_LIVE_RESTORE_FS *lr_fs,
-  const char *name, uint32_t flags, WTI_LIVE_RESTORE_FILE_HANDLE *lr_fh, bool have_stop,
+  const char *name, uint32_t flags, WTI_LIVE_RESTORE_FILE_HANDLE *lr_fh,
   bool dest_exist, bool source_exist)
 {
     WT_RET(__live_restore_fs_open_in_destination(lr_fs, session, lr_fh, name, flags, !dest_exist));
-    if (have_stop || __wti_live_restore_migration_complete(session) || !source_exist)
+    if (!source_exist)
         return (0);
 
+    /*
+     * In theory we have already completed the migration for this file which would mean this open
```

Comment:
I just realised I don't understand how the background threads and app threads access the same file at the same time. How is it possible for us to be opening a file while the background thread is already migrating. Doesn't this imply we can have multiple file handles open on a single file?

Reply:
My understanding is that the dhandles are where we achieve many users to single file handle, it's either that or WT_FH* (not WT_FILE_HANDLE*).  So no we never have multiple handles open on a single file at the same time, that would break every assumption we make in live restore.